frontend/src/pages/PipelineDetailsV1.tsx

```diff
@@ -136,6 +136,15 @@ const PipelineDetailsV1: React.FC<PipelineDetailsV1Props> = ({
   const createVersionUrl = () => {
     return selectedVersion!.code_source_url!;
   };
+  const createVersionDescription = () => {
+    if (selectedVersion) {
+      return selectedVersion.description;
+    }
+    if (pipeline && pipeline.default_version) {
+      return pipeline.default_version.description;
+    }
```

Comment:
Should we show the default version description if selected version description is not available? I think we should remove this part, so that when selected version description is not available, we fall back to pipeline description. How do you think?

Reply:
So I believe this is what the current code does. 

Currently:
- Show selected version description.
- If not available, show pipeline default version description.
- If not available, show pipeline description.

Logic currently is split between `createVersionDescription` and line 207 - 2017. The reason for this split is to make explicit to the user we are showing a Version description vs Pipeline description. We could merge them if wanted.

Reply:
Yes my understanding of current code aligns with your description here. I suggest to make the following change:


- Show selected version description.
- ~~If not available, show pipeline default version description.~~
- If not available, show pipeline description.

For example, if there are version1(default version) and version2. When user selects version2 and version2 doesn't have description, the current code will show version1's description, which can bring confusion to user. Therefore I suggest deleting Line 143-145.


> Logic currently is split between createVersionDescription and line 207 - 217. The reason for this split is to make explicit to the user we are showing a Version description vs Pipeline description. We could merge them if wanted.

This split is okay, we don't need to merge them.

Reply:
Agree with @zijianjoy

Reply:
So after testing locally and more found out what I actually implemented (which leads me to think this logic is hard).

The actual logic is:

- If selected version: Show selected version description, if not available: Show pipeline description.
- If not selected version: Show default pipeline version description, if not available: Show pipeline description

The thing is that if there is a SelectedVersion, the line in 141 will return, even if description is empty or undefined. Which means the pipeline description will be shown. Lines 143-145 won't execute unless there's not a selectedVersion.

If I remove the lines 143-145, when loading the pipeline on its default version, we wont see the version description by default. This would make it hard to find the version description for default version which may be useful to know what SHA is being used in the current version.

Lmk if it makes sense.

Reply:
Updated code to actually just use a variable as suggested by @Bobgy . Hoping it also helps with understanding with readibilty.

Reply:
>  This would make it hard to find the version description for default version which may be useful to know as in the described user usage it includes what SHA is being used to build the version of the pipeline.

I think this goal is what makes the logic hard to implement. I suggest to describe user usage that they include original SHA in the `pipeline description`, other than only in `default version description`. The main reason is that user doesn't know whether the `version description` on UI is the current version description, or the default version description. We wrap this logic in the implementation detail, and it will cause confusion to user because user sees `Version Description` title in either case.

On the side note, if showing both pipeline description and version description at the same time helps simplifying the logic and achieving the goal of sharing useful information to users, feel free to do so.